src/js/stores/BallotStore.js

```diff
@@ -127,13 +133,26 @@ class BallotStore extends FluxMapStore {
       case "voterAddressSave":
         key = action.res.google_civic_election_id;
         newBallot[key] = action.res;
+        // console.log("BallotStore, voterAddressSave, key: " + key);
 
+        // We need to refresh the ballot here after the address is changed. If we don't, it is possible to change
+        // the address and be left with the ballot for the prior address.
+        BallotActions.retrieve();
```

Comment:
I don't think that you should change this either. The voterAddressSave endpoint returns the new ballot from the new address, so the currently the BallotStore is saving the new ballot information. There's no need to retrieve the ballot again.